lib/fetch/util.js

```diff
@@ -200,12 +212,24 @@ function setRequestReferrerPolicyOnRedirect (request, actualResponse) {
 
   // 1. Let policy be the result of executing § 8.1 Parse a referrer policy
   // from a Referrer-Policy header on actualResponse.
-  // TODO:  https://w3c.github.io/webappsec-referrer-policy/#parse-referrer-policy-from-header
-  const policy = ''
+
+  // 8.1 Parse a referrer policy from a Referrer-Policy header
+  // 1. Let policy-tokens be the result of extracting header list values given `Referrer-Policy` and response’s header list.
+  const { headersList } = actualResponse
+  // 2. Let policy be the empty string.
+  // 3. For each token in policy-tokens, if token is a referrer policy and token is not the empty string, then set policy to token.
+  // 4. Return policy.
+  const token = headersList.get('referrer-policy') ?? ''
```

Comment:
It should pick the first valid one, this is similar to how the `integrity` option on RequestInit works.

There has to be some validation that the first referrer policy is actually a referrer policy. Otherwise a referrer policy like `invalid, origin` wouldn't work.

Reply:
Addressed and added tests in [683ef3a](https://github.com/nodejs/undici/pull/1717/commits/683ef3aaa32577d43d82a6e8f27c9afa58c94d1e)